app/services/v1/user_session_schedule_service.rb

```diff
@@ -79,6 +79,7 @@ def schedule_until(date_of_schedule, next_session)
 
     if date_of_schedule.past?
       user_intervention.update!(status: :in_progress)
+      next_user_session.update!(scheduled_at: date_of_schedule)
```

Comment:
In lines 82 and 88 you do the same so take it to another place where the condition will not be checked